sdk/openai/inference/src/clients/azure_openai_client.rs

```diff
@@ -0,0 +1,151 @@
+    ) -> Result<Self> {
+        let endpoint = Url::parse(endpoint.as_ref())?;
+
+        let options = client_options.unwrap_or_default();
+
+        let auth_policy = Arc::new(BearerTokenCredentialPolicy::new(credential, DEFAULT_SCOPE));
+        let version_policy: Arc<dyn Policy> = options.api_service_version.clone().into();
+        let per_call_policies: Vec<Arc<dyn Policy>> = vec![auth_policy, version_policy];
+
+        let pipeline = super::new_pipeline(per_call_policies, options.client_options.clone());
+
+        Ok(AzureOpenAIClient {
+            endpoint,
+            pipeline,
+            options,
+        })
+    }
+
+    /// Creates a new [`AzureOpenAIClient`] using a key credential
+    ///
+    /// # Parameters
+    /// * `endpoint` - The full URL of the Azure OpenAI resource endpoint.
+    /// * `secret` - The key creadential used for authentication. Passed as header parameter in the request.
+    /// * `client_options` - Optional configuration for the client. The [`AzureServiceVersion`](crate::options::AzureServiceVersion) can be provided here.
+    ///
+    /// # Example
+    /// ```no_run
+    /// use azure_openai_inference::clients::{AzureOpenAIClient, AzureOpenAIClientMethods};
+    ///
+    /// let endpoint = std::env::var("AZURE_OPENAI_ENDPOINT").expect("Set AZURE_OPENAI_ENDPOINT environment variable");
+    /// let secret = std::env::var("AZURE_OPENAI_KEY").expect("Set AZURE_OPENAI_KEY environment variable");
+    /// let client = AzureOpenAIClient::with_key_credential(
+    ///     endpoint,
+    ///     secret,
+    ///     None,
+    /// ).unwrap();
+    /// ```
+    pub fn with_key_credential(
+        endpoint: impl AsRef<str>,
+        secret: impl Into<String>,
+        client_options: Option<AzureOpenAIClientOptions>,
+    ) -> Result<Self> {
+        let endpoint = Url::parse(endpoint.as_ref())?;
+
+        let options = client_options.unwrap_or_default();
+
+        let auth_policy: Arc<dyn Policy> = AzureKeyCredential::new(secret).into();
+        let version_policy: Arc<dyn Policy> = options.api_service_version.clone().into();
+        let per_call_policies: Vec<Arc<dyn Policy>> = vec![auth_policy, version_policy];
+
+        let pipeline = super::new_pipeline(per_call_policies, options.client_options.clone());
+
+        Ok(AzureOpenAIClient {
+            endpoint,
+            pipeline,
+            options,
+        })
+    }
+}
+
+impl AzureOpenAIClientMethods for AzureOpenAIClient {
+    /// Returns the endpoint [`Url`] of the client.
+    fn endpoint(&self) -> &Url {
+        &self.endpoint
+    }
+
+    /// Returns a new instance of the [`ChatCompletionsClient`] using an [`AzureOpenAIClient`] underneath.
+    fn chat_completions_client(&self) -> ChatCompletionsClient {
+        ChatCompletionsClient::new(Box::new(self.clone()))
+    }
+}
+
+impl BaseOpenAIClientMethods for AzureOpenAIClient {
+    fn base_url(&self, deployment_name: Option<&str>) -> azure_core::Result<Url> {
```

Comment:
As a caller, I would wonder what the difference is between `endpoint` and `base_url`. Public API docs will be required before you can ship, but I would describe at least these two methods now.